libexec/apply_catalog.rb

```diff
@@ -39,7 +40,14 @@
 Puppet[:prerun_command] = nil
 Puppet[:postrun_command] = nil
 
-env = Puppet.lookup(:environments).get('production')
+Puppet[:default_file_terminus] = :file_server
+
+moduledir = Dir.mktmpdir
+plugins = Tempfile.new('plugins.tar.gz')
+File.binwrite(plugins, Base64.decode64(args['plugins']))
+Puppet::ModuleTool::Tar.instance.unpack(plugins, moduledir, Etc.getlogin)
```

Comment:
I thought weren't going to pass this as a task param. I guess this is fine until it causes problems. We probably need to make sure we don't use environment variables for the catalog_apply_task though.

Reply:
Oh yeah, this could be a file upload, which would probably be faster for ssh.

Reply:
Tempted to leave that as an improvement just to get this out faster. But I can experiment with it a bit.

Reply:
yeah I think we'll want to do something else there but lets ticket it later.